std/debug/failing_allocator.zig

```diff
@@ -10,6 +10,7 @@ pub const FailingAllocator = struct {
     internal_allocator: *mem.Allocator,
     allocated_bytes: usize,
     freed_bytes: usize,
+    allocations: usize,
```

Comment:
For changes to this struct, please see my review comments on #2310

Reply:
> So it would introduce a new field, allocations, which increments in realloc when old_mem.len == 0. 
> Meanwhile, deallocations would increment in both realloc and shrink when new_size == 0.

This seems odd to me: why would `allocations` not increment on grow when `deallocations` does increment on shrink?